src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/SwitchStatementHighlighter.cs

```diff
@@ -26,12 +27,14 @@ internal class SwitchStatementHighlighter : AbstractKeywordHighlighter<SwitchSta
                 {
                     spans.Add(label.Keyword.Span);
                     spans.Add(EmptySpan(label.ColonToken.Span.End));
-                }
 
-                foreach (var statement in switchSection.Statements)
-                {
-                    HighlightRelatedKeywords(statement, spans);
+                    if (label is CasePatternSwitchLabelSyntax patternLabel && patternLabel.WhenClause != null)
+                    {
+                        spans.Add(patternLabel.WhenClause.WhenKeyword.Span);
+                    }
                 }
+
+                HighlightRelatedKeywords(switchSection, spans, true, true);
```

Comment:
named parameters for contants.

Reply:
similar code is in https://github.com/dotnet/roslyn/blob/6847f1e5a909395aae9456e8f366cbf4deb86b69/src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/LoopHighlighter.cs#L57 fix it there as well?

Reply:
Yes please.